operations/fetch.go

```diff
@@ -248,6 +248,17 @@ type cloneOptions struct {
 }
 
 func clone(opts cloneOptions) error {
+	// Check repository existence if no token is provided
+	if opts.token == "" {
+		resp, err := http.Get(thirdparty.FormGitURLForApp("github.com", opts.owner, opts.repository, opts.token))
+		if err != nil {
+			return errors.Errorf("failed to check if %s/%s exists: %v", opts.owner, opts.repository, err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			return errors.Errorf("repository does not exist or is private an no token was provided: %d", resp.StatusCode)
```

Comment:
can we print the name of the repo in this error too

Reply:
Yes, good call. Thank you!